quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/QuarkusCli35to315UpdateIT.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.ts.quarkus.cli.update;
+
+import static io.quarkus.test.util.QuarkusCLIUtils.getDependencies;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.model.Dependency;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+
+public class QuarkusCli35to315UpdateIT extends AbstractQuarkusCliUpdateIT {
+    private static final DefaultArtifactVersion oldVersion = new DefaultArtifactVersion("3.5");
+    private static final DefaultArtifactVersion newVersion = new DefaultArtifactVersion("3.15");
+
+    public QuarkusCli35to315UpdateIT() {
+        super(oldVersion, newVersion);
+    }
+
+    /**
+     * Test that updating from Quarkus 3.5 to 3.15 correctly transforms
+     * the 'quarkus-rest-client-reactive-jackson' extension to 'quarkus-rest-client-jackson',
+     * and does not incorrectly transform it to 'quarkus-resteasy-client-jackson'.
+     * This test addresses a specific issue where the update recipes could apply in the wrong order,
+     * causing incorrect dependency transformations. More info here:https://github.com/quarkusio/quarkus/issues/44183
+     *
+     */
+
+    @Tag("https://github.com/quarkusio/quarkus/issues/44183")
+    @Test
+    public void testUpdateFrom35to315() throws XmlPullParserException, IOException {
```

Comment:
IMHO a better suited name would be something like "testCorrectRecipiesOrder" or something like that.
Goal is not to test update from 3.5 to 3.15, That is only known case, where the recipies order issue manifests